### PR TITLE
init test-js

### PR DESCRIPTION
**Pull request by mattlockyer**
@fxfactorial the initial test-js branch.

Please review and let me know if there are any changes before I proceed with more validator test.

### CODE REVIEW

File: test-automation/api-tests/package.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (fxfactorial) [inline test-automation/api-tests/package.json]: harmny-sdk is nodejs 10.0 min, so please enforce that here engineStrict key, also please open pr to harmony-js sdk itself to enforce the nodejs 10 min requirement as well:

File: test-automation/api-tests/.gitignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+.note
```
Comment (fxfactorial) [inline test-automation/api-tests/.gitignore]: new .gitignore not needed, please add to rootmost one: https://github.com/harmony-one/harmony-ops/blob/master/.gitignore

File: test-automation/api-tests/api-tests [hunk 3]
```diff
@@ -0,0 +1 @@
+/home/matt/Projects/harmony/harmony-ops/test-automation/api-tests
```
Comment (fxfactorial) [inline test-automation/api-tests/api-tests]: what is purpose of this file

File: test-automation/api-tests/test.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+
+const { Harmony } = require('@harmony-js/core');
+
+// import or require settings
+const { ChainID, ChainType } = require('@harmony-js/utils');
+const { StakingFactory } = require('@harmony-js/staking');
+
+const URL_TESTNET = `https://api.s0.b.hmny.io`;
```
Comment (fxfactorial) [inline test-automation/api-tests/test.js]: I think these better as env vars so easy from shell to set

**Comment by mattlockyer** [comment]
@fxfactorial a bunch of new additions, some comments on what is working vs. what is not.

Missing from the JS SDK is create/edit validator.

Missing from this test suite is the loading of an account with local net funds, delegating and confirming that delegation (among other things).

Mostly, it's still a stubbed out test suite calling the JS SDK and not fully testing end to end.

**Pull request merged**